src/Core/src/Platform/Android/MauiApplication.cs

```diff
@@ -40,4 +36,18 @@ void ConfigureNativeServices(HostBuilderContext ctx, IServiceCollection services
 		{
 		}
 	}
+
+	public abstract class MauiApplication : Android.App.Application
```

Comment:
This is the real magic. Once the services and application is created, we set the statics here since there can ever only be a single running Android app. We need a non-generic base type because we need a way to get the `Current` property. If we had a generic, then we also will need to pass the `TStartup` everywhere.